src/libraries/System.Net.Http/tests/FunctionalTests/ResponseStreamConformanceTests.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using System.IO.Tests;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Http.Functional.Tests
+{
+    public sealed class Http1CloseResponseStreamConformanceTests : ResponseConnectedStreamConformanceTests
+    {
+        protected override string GetResponseHeaders() => "HTTP/1.1 200 OK\r\nConnection: close\r\n\r\n";
+
+        protected override async Task<StreamPair> CreateConnectedStreamsAsync()
+        {
+            StreamPair pair = await base.CreateConnectedStreamsAsync();
+            Assert.False(pair.Stream2.CanWrite);
+            Assert.True(pair.Stream2.CanRead);
+            return pair;
+        }
+    }
+
+    public sealed class Http1RawResponseStreamConformanceTests : ResponseConnectedStreamConformanceTests
+    {
+        protected override string GetResponseHeaders() => "HTTP/1.1 101 Switching Protocols\r\n\r\n";
+
+        protected override async Task<StreamPair> CreateConnectedStreamsAsync()
+        {
+            StreamPair pair = await base.CreateConnectedStreamsAsync();
+            Assert.True(pair.Stream2.CanWrite);
+            Assert.True(pair.Stream2.CanRead);
+            return pair;
+        }
+    }
+
+    public sealed class Http1ContentLengthResponseStreamConformanceTests : ResponseStandaloneStreamConformanceTests
+    {
+        protected override async Task WriteResponseAsync(Stream responseStream, byte[] bodyData)
+        {
+            await responseStream.WriteAsync(Encoding.ASCII.GetBytes($"HTTP/1.1 200 OK\r\nContent-Length: {bodyData.Length}\r\n\r\n"));
+            await responseStream.WriteAsync(bodyData);
+        }
+    }
+
+    public sealed class Http1ChunkedResponseStreamConformanceTests : ResponseStandaloneStreamConformanceTests
+    {
+        protected override async Task WriteResponseAsync(Stream responseStream, byte[] bodyData)
+        {
+            await responseStream.WriteAsync(Encoding.ASCII.GetBytes($"HTTP/1.1 200 OK\r\nTransfer-Encoding: chunked\r\n\r\n{bodyData.Length:X}\r\n"));
+            await responseStream.WriteAsync(bodyData);
+            await responseStream.WriteAsync(Encoding.ASCII.GetBytes("\r\n"));
+            await responseStream.WriteAsync(Encoding.ASCII.GetBytes("0\r\n\r\n"));
+        }
+    }
+
+    public abstract class ResponseConnectedStreamConformanceTests : ConnectedStreamConformanceTests
+    {
+        protected override Type UnsupportedConcurrentExceptionType => null;
+        protected override bool UsableAfterCanceledReads => false;
```

Comment:
Is this something we should fix?

Reply:
I'm not sure how feasible it is in general.  If this were an SSL connection for example, we could end up reading part of a message (e.g. a header) and then get canceled while reading the rest, which is why SslStream itself doesn't provide this guarantee.  Same goes for, say, chunked responses.  In general I also don't think it's very important: if you're canceling, you're almost certainly going away.

Reply:
I agree it's not really important, and I'm fine saying we just don't support this.

That said I think we *could* support this if we wanted to; it would just require some hardening in places like SslStream and the chunked response logic.